components/ImageCard.js

```diff
@@ -0,0 +1,176 @@
+import {useState, useEffect} from "react";
+import Image from "next/legacy/image";
+import styled from "styled-components";
+import {useContext} from "react";
+import {ImageContext} from "./ImageContext";
+
+export default function ImageCard() {
+  const [dataArray, setDataArray] = useState([]);
+  const [isLoading, setLoading] = useState(true);
+  const [displayImage, setDisplayImage] = useState(null);
+  const [reload, setReload] = useState(true);
+  const [random, setRandom] = useState(0);
+  const {addFavorite} = useContext(ImageContext);
+
+  useEffect(() => {
+    setLoading(true);
+    const getImages = async () => {
+      try {
+        const response = await fetch(
+          "https://api.europeana.eu/record/v2/search.json?media=true&profile=minimal&query=Painting&theme=art&reusability=open&qf=IMAGE_COLOUR:%22TRUE%22&rows=100&wskey=hanozzle"
+        );
+        if (response.ok) {
+          const serverData = await response.json();
+          setDataArray(serverData.items);
+          setLoading(false);
+          // setDisplayImage(randomizeImage(serverData.items));
+          setReload(!reload);
+        } else {
+          throw new Error(
+            "Fetch fehlgeschlagen mit Status: ${response.status}"
+          );
+        }
+      } catch (error) {
+        console.log(error);
+        alert(error.message);
+      }
+    };
+    getImages();
+  }, []);
+
+  useEffect(() => {
+    setRandom(Math.floor(Math.random() * dataArray.length));
+  }, [reload]);
+
+  useEffect(() => {
+    setDisplayImage(dataArray[random]);
+  }, [random]);
+
+  // function randomizeImage(dataArray) {
+  //   const randomIndex = Math.floor(Math.random() * dataArray.length);
+  //   const randomImage = dataArray[randomIndex];
+  //   return randomImage;
+  // }
+
+  if (isLoading) return <p>Loading...</p>;
+  if (!dataArray) return <p>No data</p>;
+  return (
+    <div>
+      {displayImage ? (
+        <>
+          <ImageContainer>
+            <Image
+              key={displayImage.index}
+              alt={displayImage.title[0]}
+              src={displayImage.edmPreview[0]}
+              width={500}
+              height={300}
+            />
+          </ImageContainer>
+          <Details>
+            <p>{displayImage.title[0]}</p>
+            <p>{displayImage.dcCreator}</p>
+          </Details>
+        </>
+      ) : (
+        ""
+      )}
```

Comment:
You could als work with a logical and:

```suggestion
      {displayImage && (
        <>
          <ImageContainer>
            <Image
              key={displayImage.index}
              alt={displayImage.title[0]}
              src={displayImage.edmPreview[0]}
              width={500}
              height={300}
            />
          </ImageContainer>
          <Details>
            <p>{displayImage.title[0]}</p>
            <p>{displayImage.dcCreator}</p>
          </Details>
        </>
      )}
```

Reply:
Done. Is a bit clearer.